src/fluree/server/handlers/transact.clj

```diff
@@ -102,13 +103,16 @@
                                 :body   {:error err-message}}}))))
 
 (defhandler default
-  [{:keys [fluree/conn fluree/consensus fluree/watcher fluree/broadcaster credential/did raw-txn]
+  [{:keys [fluree/conn fluree/consensus fluree/watcher fluree/broadcaster credential/did fluree/opts raw-txn]
     {:keys [body]} :parameters}]
-  (let [txn-context    (ctx-util/txn-context body)
-        ledger-id      (transact-api/extract-ledger-id body)]
+  (let [txn            (if (= :sparql (:format opts))
```

Comment:
I think you should use the function you just defined in db here.

Also, maybe not for this pr, but I think we should eventually write a function that can extract the ledger id directly from a sparql query, or move all sparql conversion outside of the db library here. (This is just a gut level thought. I haven't thought through all of the implications yet)